apps/backoffice-v2/src/common/components/organisms/Filters/Filters.Input.tsx

```diff
@@ -0,0 +1,19 @@
+import { MultiselectFilterInput } from '@/common/components/organisms/Filters/filter-inputs/MultiSelectFilterInput';
+import { IFilterInputParams } from '@/common/components/organisms/Filters/interfaces';
+import { TFilterInputType } from '@/common/components/organisms/Filters/types';
+import { FunctionComponent, useMemo } from 'react';
+
+const getFilterInputComponent = (
+  type: TFilterInputType,
+): React.ComponentType<IFilterInputParams> => {
```

Comment:
If you remove the return type TypeScript could understand on its own the props. Or at least make use of `Extract`.